libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/lifecycle/UIViewModel.kt

```diff
@@ -21,10 +21,15 @@ import kotlinx.coroutines.launch
  * Behaviors have a lifecycle, contain state, and process actions. [UIComponent] will respond
  */
 @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
-abstract class UIViewModel<State, Action>(initialState: State) : MapboxNavigationObserver {
+abstract class UIViewModel<State, Action>(
+    mutableState: MutableStateFlow<State>
```

Comment:
@tomaszrybakiewicz is this a residual change when resolving the circular dependency?

We don't have a reason to do this yet so this should be reverted

Reply:
Let's merge though, I can follow up on this